substrate/frame/child-bounties/src/migration.rs

```diff
@@ -0,0 +1,163 @@
+use super::*;
+// use codec::{Decode, Encode};
+use core::marker::PhantomData;
+use frame_support::{
+	storage_alias,
+	traits::{Get, UncheckedOnRuntimeUpgrade},
+};
+
+#[cfg(feature = "try-runtime")]
+use alloc::vec::Vec;
+#[cfg(feature = "try-runtime")]
+use frame_support::ensure;
+
+pub mod v1 {
+	use super::*;
+
+	pub struct MigrateToV1Impl<T>(PhantomData<T>);
+
+	#[storage_alias]
+	type ChildBountyDescriptions<T: Config + pallet_bounties::Config> = StorageMap<
+		Pallet<T>,
+		Twox64Concat,
+		BountyIndex,
+		BoundedVec<u8, <T as pallet_bounties::Config>::MaximumReasonLength>,
+	>;
+
+	#[storage_alias]
+	type ChildrenCuratorFees<T: Config> =
+		StorageMap<Pallet<T>, Twox64Concat, BountyIndex, BalanceOf<T>, ValueQuery>;
+
+	impl<T: Config> UncheckedOnRuntimeUpgrade for MigrateToV1Impl<T> {
+		fn on_runtime_upgrade() -> frame_support::weights::Weight {
+			// increment reads/writes after the action
+			let mut reads = 0u64;
+			let mut writes = 0u64;
+
+			let mut old_bounty_ids = Vec::new();
+			// first iteration collect all existing ids not to mutate map as we iterate it
+			for (parent_bounty_id, old_child_bounty_id) in ChildBounties::<T>::iter_keys() {
+				reads += 1;
+				old_bounty_ids.push((parent_bounty_id, old_child_bounty_id));
+			}
+
+			log::info!(
+				target: LOG_TARGET,
+				"Migrating {} child bounties",
+				old_bounty_ids.len(),
+			);
+
+			for (parent_bounty_id, old_child_bounty_id) in old_bounty_ids {
+				// assign new child bounty id
+				let new_child_bounty_id = ParentTotalChildBounties::<T>::get(parent_bounty_id);
+				reads += 1;
+				ParentTotalChildBounties::<T>::insert(
+					parent_bounty_id,
+					new_child_bounty_id.saturating_add(1),
+				);
+				writes += 1;
+
+				log::info!(
+					target: LOG_TARGET,
+					"Remapped parent bounty {} child bounty id {}->{}",
+					parent_bounty_id,
+					old_child_bounty_id,
+					new_child_bounty_id,
+				);
+
+				let bounty_description =
+					v1::ChildBountyDescriptions::<T>::take(old_child_bounty_id);
+				writes += 1;
+				let bounty_curator_fee = v1::ChildrenCuratorFees::<T>::take(old_child_bounty_id);
+				writes += 1;
+				let child_bounty = ChildBounties::<T>::take(parent_bounty_id, old_child_bounty_id);
+				writes += 1;
+
+				// should always be some
+				if let Some(taken) = child_bounty {
+					ChildBounties::<T>::insert(parent_bounty_id, new_child_bounty_id, taken);
+					writes += 1;
+				}
+				if let Some(bounty_description) = bounty_description {
+					super::super::ChildBountyDescriptions::<T>::insert(
+						parent_bounty_id,
+						new_child_bounty_id,
+						bounty_description,
+					);
+					writes += 1;
+				}
+				super::super::ChildrenCuratorFees::<T>::insert(
+					parent_bounty_id,
+					new_child_bounty_id,
+					bounty_curator_fee,
+				);
+				writes += 1;
```

Comment:
the read/writes calculation are not correct, you do not consider the overlay (cache) layer

Reply:
I think that would make things kind of complex, is it a high impact if we return little higher cost for migration? The way I see it, worst case I see it, one block will have little more room for migration than needed, and next block it's already done.

Say, we get `ParentTotalChildBounties::<T>::get` multiple times for the same bounty, we'd need code to account where it is fetched from the cache (store in another map if we've seen this parent bounty already) which would IMO obstruct the main code of migration.